pkg/crc/machine/stop_test.go

```diff
@@ -6,11 +6,71 @@ import (
 	"testing"
 
 	crcConfig "github.com/crc-org/crc/v2/pkg/crc/config"
+	"github.com/crc-org/crc/v2/pkg/crc/machine/fakemachine"
 	"github.com/crc-org/crc/v2/pkg/crc/machine/state"
 	crcOs "github.com/crc-org/crc/v2/pkg/os"
 	"github.com/stretchr/testify/assert"
 )
 
+func TestStop_WhenVMRunning_ThenShouldStopVirtualMachine(t *testing.T) {
+	// Given
+	crcConfigStorage := crcConfig.New(crcConfig.NewEmptyInMemoryStorage(), crcConfig.NewEmptyInMemorySecretStorage())
+	crcConfigStorage.AddSetting(crcConfig.NetworkMode, "user", crcConfig.ValidateBool, crcConfig.SuccessfullyApplied,
+		"network-mode")
+	_, err := crcConfigStorage.Set(crcConfig.NetworkMode, "true")
+	assert.NoError(t, err)
+	virtualMachine := fakemachine.NewFakeVirtualMachine(false, false)
+	client := newClientWithVirtualMachine("fake-virtual-machine", false, crcConfigStorage, virtualMachine)
+
+	// When
+	clusterState, stopErr := client.Stop()
+
+	// Then
+	assert.NoError(t, stopErr)
+	assert.Equal(t, clusterState, state.Stopped)
+	assert.Equal(t, virtualMachine.IsStopped, true)
+	assert.Equal(t, virtualMachine.FakeSSHClient.LastExecutedCommand, "sudo -- sh -c 'crictl stop $(crictl ps -q)'")
```

Comment:
In my opinion, the fact that `crictl stop` is the last command we run in the VM should be an implementation detail, it can change at any time if  we refactor this code, if we add new things to the bundle which needs to be stopped, ...

Reply:
Thanks for review, I'll remove this assertion.